java-checks/src/test/files/checks/StringCallsBeyondBoundsCheck.java

```diff
@@ -0,0 +1,73 @@
+class A {
+  void f() {
+    String str = "Hello, world!";
+    char[] array = new char[str.length()];
+    int zero = 0, minusOne = -1, any = System.currentTimeMillis();
+
```

Comment:
I think following pattern should be detected 
```
    s.charAt(s.length());
```
even if we don't statically know the lenght of `s` . All your tests assume that we statically know the length of the string (which is fine), but we should also cover the cases where `length()` is a parameter to the call on the same symbol